djstripe/views.py

```diff
@@ -174,23 +180,20 @@ def post(self, request, *args, **kwargs):
                 # then we force the proration of his current plan and use it towards the upgraded plan,
                 # no matter what PRORATION_POLICY is set to.
                 if PRORATION_POLICY_FOR_UPGRADES:
+                    # TODO: Needs refactor
                     current_subscription_amount = customer.current_subscription.amount
-                    selected_plan_name = form.cleaned_data["plan"]
-                    selected_plan = next(
-                        (plan for plan in PLAN_LIST if plan["plan"] == selected_plan_name)
-                    )
-                    selected_plan_price = selected_plan["price"]
-
-                    if not isinstance(selected_plan["price"], decimal.Decimal):
-                        selected_plan_price = selected_plan["price"] / decimal.Decimal("100")
-
-                    # Is it an upgrade?
+                    selected_plan_id = form.cleaned_data["plan"]
+                    selected_plan = Plan.objects.get(stripe_id=selected_plan_id)
+                    selected_plan_price = selected_plan.amount
+                    if not isinstance(selected_plan_price, decimal.Decimal):
+                        selected_plan_price = selected_plan_price / decimal.Decimal("100")
+                    """ Is it an upgrade """
                     if selected_plan_price > current_subscription_amount:
-                        customer.subscribe(selected_plan_name, prorate=True)
+                        customer.subscribe(selected_plan_id, prorate=True)
                     else:
-                        customer.subscribe(selected_plan_name)
+                        customer.subscribe(selected_plan_id)
                 else:
-                    customer.subscribe(form.cleaned_data["plan"])
+                    customer.subscribe(form.cleaned_data.get("plan"))
```

Comment:
subscribing to Nonetype will throw errors that aren't straightforward to troubleshoot.. Best to throw a keyerror here as before.